lib/foreman/renderer.rb

```diff
@@ -99,7 +113,10 @@ def snippet(name, options = {})
         begin
           return unattended_render(template, nil, options[:variables] || {})
         rescue => exc
-          raise "The snippet '#{name}' threw an error: #{exc}"
+          # fixes the backtrace but ideally should raise better error message with Foreman rendering exception, same everywhere in this file
```

Comment:
Either using a different exception or leaving it as-is (but minus the comment, which I don't think is really needed) is fine, provided that the current exception captures the original message. If using a different exception class then it should probably derive from Foreman::WrappedException?